//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/enclave.proto";

message In6MacroTestInput {
  optional bytes ipv6_address = 1;
}

extend EnclaveInput {
  optional In6MacroTestInput in6_macro_test_input = 256412257;
}

message MacroResults {
  optional bool unspecified = 2;
  optional bool loopback = 3;
  optional bool multicast = 4;
  optional bool link_local = 5;
  optional bool site_local = 6;
  optional bool v4_mapped = 7;
  optional bool v4_compat = 8;
  optional bool multicast_node_local = 9;
  optional bool multicast_link_local = 10;
  optional bool multicast_site_local = 11;
  optional bool multicast_org_local = 12;
  optional bool multicast_global = 13;
}

extend EnclaveOutput {
  optional MacroResults in6_macro_test_output = 256412258;
}
